### PR TITLE
[ocm-clusters] Create ROSA classic privatelink and public using ROSA CLI in a Job

### PR DESCRIPTION
**Pull request by OliviaHY**
#### What:
Adding ability to create ROSA classic cluster through generate script and calling it from a Job. It replace the original way of calling OCM API for consistency.
#### Validation:
Created ROSA privatelink cluster locally.

### CODE REVIEW

File: reconcile/utils/ocm/products.py [hunk 1]
```diff
@@ -329,18 +333,43 @@ def create_cluster(
...
-                logging.info(ocm_spec)
-                return
-        ocm.post(api, ocm_spec, params)
+        if not isinstance(cluster.spec, ROSAClusterSpec):
+            # make mypy happy
+            return
+
+        if self.rosa_session_builder is None:
+            raise Exception(
+                "OCMProductHypershift is not configured with a rosa session builder"
```
Comment (geoberle) [inline reconcile/utils/ocm/products.py:342]:
```suggestion
                "OCMProductROSA is not configured with a rosa session builder"
```

File: reconcile/utils/rosa/session.py [hunk 2]
```diff
@@ -129,6 +129,25 @@ def create_hcp_cluster(
...
+        check_interval_seconds: int = 10,
+        timeout_seconds: int = 300,
+    ) -> RosaCliResult:
+        """
+        Create a ROSA cluster in the OCM org and AWS account of this session.
+        If dry-run is provided, cluster creation is simulated and invalid configuration
+        data is highlighted as errors.
+        """
+        return self.cli_execute(
+            cmd=rosa_hcp_creation_script(cluster_name, spec, dry_run),
```
Comment (geoberle) [inline reconcile/utils/rosa/session.py:146]: rename the function so it does not have hcp in the name

File: reconcile/utils/rosa/session.py [hunk 3]
```diff
@@ -124,7 +124,26 @@ def create_hcp_cluster(
         data is highlighted as errors.
         """
         return self.cli_execute(
-            cmd=rosa_hcp_creation_script(cluster_name, spec, dry_run),
+            cmd=generate_rosa_creation_script(cluster_name, spec, dry_run),
+            check_interval_seconds=check_interval_seconds,
+            timeout_seconds=timeout_seconds,
+        )
+
+    def create_rosa_cluster(
```
Comment (geoberle) [inline reconcile/utils/rosa/session.py:132]: i think there is no difference left between this function and `create_hcp_cluster`. how about we delete `create_hcp_cluster`?

**Review by geoberle — APPROVED** [review]

File: reconcile/utils/rosa/session.py (hunk 3 above)
Comment (OliviaHY) [reply reconcile/utils/rosa/session.py:132]: Absolutely!

**Pull request merged**